tests/unit/synapseclient/core/multithread_download/unit_test_download_threads.py

```diff
@@ -147,6 +147,48 @@ def test_pre_signed_url_expiration_time():
     assert expected == download_threads._pre_signed_url_expiration_time(url)
 
 
+def test_pre_signed_url_expires_seconds_since_epoch():
+    # GIVEN a pre signed URL with an expiration date of: Jul 22 2013 20:12:07 in seconds since epoch UTC
+    url = (
+        "https://s3.amazonaws.com/examplebucket/test.txt"
+        "?X-Amz-Algorithm=AWS4-HMAC-SHA256"
+        "&X-Amz-Credential=your-access-key-id/20130721/us-east-1/s3/aws4_request"
+        "&Expires=1374523927"
+        "&X-Amz-SignedHeaders=host"
+        "&X-Amz-Signature=signature-value"
+    )
+
+    # WHEN I get the expiration time from the pre signed URL
+    expiration_time = download_threads._pre_signed_url_expiration_time(url)
+
+    # THEN I expect the expiration time to be Jul 22 2013 20:12:07
+    expected = datetime.datetime(
+        year=2013, month=7, day=22, hour=20, minute=12, second=7
+    )
+    assert expected == expiration_time
+
+
+def test_pre_signed_url_no_expiration_time():
+    # GIVEN a pre signed URL with no expiration date
+    url = (
+        "https://s3.amazonaws.com/examplebucket/test.txt"
+        "?X-Amz-Algorithm=AWS4-HMAC-SHA256"
+        "&X-Amz-Credential=your-access-key-id/20130721/us-east-1/s3/aws4_request"
+        "&X-Amz-SignedHeaders=host"
+        "&X-Amz-Signature=signature-value"
+    )
+
+    # WHEN I get the expiration time from the pre signed URL
+    with pytest.raises(ValueError) as expiration_error:
```

Comment:
NIT: Rather than doing the `assert` to match the strings below, you can take care of that in this line.
```suggestion
    with pytest.raises(ValueError, match="pre_signed_url does not contain*") as expiration_error:
```
[source](https://stackoverflow.com/questions/23337471/how-do-i-properly-assert-that-an-exception-gets-raised-in-pytest/56569533#56569533)